central/processlisteningonport/datastore/datastore_impl.go

```diff
@@ -63,40 +85,44 @@ func (ds *datastoreImpl) AddProcessListeningOnPort(
 	}
 
 	normalizedPLOPs, completedInBatch := normalizePLOPs(portProcesses)
-
-	// TODO ROX-14376: The next two calls, fetchIndicators and fetchExistingPLOPs, have to
-	// be done in a single join query fetching both ProcessIndicator and needed
-	// bits from PLOP.
-	indicatorsMap, indicatorIds, err := ds.fetchIndicators(ctx, normalizedPLOPs...)
-	if err != nil {
-		return err
+	allPLOPs := append(normalizedPLOPs, completedInBatch...)
+	indicatorIds := getIndicatorIdsForPlops(allPLOPs)
+
+	indicators, nonempty, err := ds.indicatorDataStore.GetProcessIndicators(ctx, indicatorIds)
+	indicatorsMap := map[string]*storage.ProcessIndicator{}
+
+	// Used to do best efforts of identifying orphaned PLOP. Note, that as
+	// usual, this could be done together with fetchExistingPLOPs in the same
+	// custom query.
+	if nonempty && err == nil {
+		for _, ind := range indicators {
+			indicatorsMap[ind.GetId()] = ind
+		}
 	}
 
+	// TODO ROX-14376: This ticket might need to be changed, but is not completely obsolete
 	existingPLOPMap, err := ds.fetchExistingPLOPs(ctx, indicatorIds)
 	if err != nil {
 		return err
 	}
 
 	plopObjects := []*storage.ProcessListeningOnPortStorage{}
 	for _, val := range normalizedPLOPs {
-		indicatorID := ""
 		var processInfo *storage.ProcessIndicatorUniqueKey
 
-		key := getPlopProcessUniqueKey(val)
-
-		if indicator, ok := indicatorsMap[key]; ok {
-			indicatorID = indicator.GetId()
-			log.Debugf("Got indicator %s: %+v", indicatorID, indicator)
-		} else {
-			countMetrics.IncrementOrphanedPLOPCounter(val.GetClusterId())
-			log.Warnf("Found no matching indicators for %s", key)
-			processInfo = val.Process
-		}
-
+		indicatorID := getIndicatorIDForPlop(val)
```

Comment:
Do you need to handle the case where this call returns an empty string?  Or does processing work as desired in that case?